login/service.go

```diff
@@ -752,6 +957,11 @@ func redirectWithError(ctx *app.LoginLoginContext, knownReferrer string, errorSt
 	return ctx.TemporaryRedirect()
 }
 
+func redirectWithErrorForAuthorize(ctx *app.AuthorizeAuthorizeContext, knownReferrer string, errorString string) error {
```

Comment:
Duplication again. Let's not use this function at all. Just return an JSON error when needed.